technico-v16/app/user/user-repairs/page.tsx

```diff
@@ -0,0 +1,135 @@
+"use client";
+
+import { useAuth } from "@/app/components/AuthContext";
+import RepairCard from "@/app/components/RepairCard";
+import { useEffect, useState } from "react";
+
+interface Repair {
+    repairId: string;
+    scheduledDate: Date;
+    type: string;
+    description: string;
+    address: string;
+    status: string;
+    cost: number;
+    userId: string;
+    propertyId: string;
+}
+
+const UserRepairs = () => {
+    const [repairs, setRepairs] = useState<Repair[]>([]);
+    const [filteredRepairs, setFilteredRepairs] = useState<Repair[]>([]);
+    const [searchQuery, setSearchQuery] = useState<string>('');
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState<string | null>(null);
+    const { isAuthenticated } = useAuth();
+
+    useEffect(() => {
+        fetchRepairs();
+    }, [isAuthenticated]);
+
+    useEffect(() => {
+        // Filter repairs dynamically based on the search query
+        const query = searchQuery.toLowerCase();
+        setFilteredRepairs(
+            repairs.filter(
+                (repair) =>
+                    repair.type.toLowerCase().includes(query) ||
+                    repair.description.toLowerCase().includes(query) ||
+                    repair.address.toLowerCase().includes(query)
+            )
+        );
+    }, [searchQuery, repairs]);
+
+    const fetchRepairs = async () => {
+        if (!isAuthenticated) return;
+
+        try {
+            setLoading(true);
+            setError(null);
+
+            const response = await fetch(`https://localhost:7118/api/Repairs/user/${localStorage.getItem('userId')}`);
```

Comment:
Consider declaring a value userId and then use it in the url for better readability